pkg/manager/member/tidbcluster_status_manager.go

```diff
@@ -125,6 +134,83 @@ func (m *TidbClusterStatusManager) syncDashboardMetricStorage(tc *v1alpha1.TidbC
 	return nil
 }
 
+// ref https://github.com/pingcap/tidb/blob/36b04d1aa01db722b3f07af759168c6b8da33801/domain/infosync/info.go#L72
+func getNoLeastTidbInfoKey(ctx context.Context, client pdapi.PDEtcdClient) (noleast []*pdapi.KeyValue, err error) {
+	kvs, err := client.Get(tidbPrefix, true /*prefix*/)
+	if err != nil {
+		return nil, perrors.AddStack(err)
+	}
+
+	infos := make(map[string]*pdapi.KeyValue)
+	ttls := make(map[string]*pdapi.KeyValue)
+
+	for _, kv := range kvs {
+		key := kv.Key
+		klog.V(4).Infof("get tidb info key %s", key)
+
+		if strings.HasSuffix(key, "ttl") {
+			ttls[key] = kv
+		}
+
+		if strings.HasSuffix(key, "info") {
+			infos[key] = kv
+		}
+	}
+
+	for key, kv := range infos {
+		if _, ok := ttls[strings.Replace(key, "/info", "/ttl", -1)]; ok {
```

Comment:
how about only replacing the suffix?

Reply:
There are no false-positive case by using `ReplaceAll` here and only replacing the suffix need more tedious code, i prefer to just keep `ReplaceAll`